src/Formula/Parsing/Delayed.hs

```diff
@@ -11,9 +11,19 @@ import Control.Monad.State (State)
 import Data.Map (Map)
 
 import LogicTasks.Helpers (reject)
+import Text.PrettyPrint.Leijen.Text (Pretty(..))
 
 newtype Delayed a = Delayed String
 
+instance Show a => Show (Delayed a) where
+  show (Delayed a) = show a
+
+instance Parse (Delayed a) where
+  parser = delayed <$> fully (many anyChar)
+
+instance Pretty a => Pretty (Delayed a) where
+  pretty (Delayed a) = pretty a
```

Comment:
Have you checked that there is no conflict between these instances and that happens in Autotool? I mean, the Autotool repo might implement "its own" `Show`, `Parse` or `Pretty` instances for `Delayed`, in which case ones implemented here might screw things up.

(In other words, while the above additions might make local sense here in that they allow to get rid of the the specialized `testTaskDelayed` below, they also change the API presented by this repo seen as a library, but then the impact on a client of this API needs to be taken into account.)

Reply:
I did not think of that. I also currently don't have a local Autotool instance set up in order to test it.

Taking a quick look at the `autotool-dev` repo revealed that there are no instance definitions for `Delayed`. Is this enough?

Reply:
I'm not sure. Part of the API of a typed library is also what expressions are *not* well-typed. That is, prior to defining those instances, certain uses would be prevented by the compiler (in Autotool or also internally in the library itself). That is, maybe we don't want someone to accidentally pass a `Delayed` thing to somewhere where it is directly printed. By not having `Show Delayed` instance, we know that the compiler will complain if someone forgets to do some explicit/alternative conversion to a string, by a dedicated function in the code instead of a simple `show` call. Now, with the `Show Delayed` instance, in the same situation the compiler would accept `show`, but it might be that it is conceptually the wrong thing to do at that point; that instead the programmer should have been warned, "Hey, you are dealing with a `Delayed` thing. You need to do something special for that." (This is all hypothetical for now.)

In any case, the instances themselves seem conceptually weird. Why is it `instance Show a => Show (Delayed a)`? Note that in a value `Delayed x :: Delayed Int`, the `x` is **not** an `Int`. It is a `String`. So,
1. the definition might as well be `show (Delayed x) = x`,
2. the constraint `Show a =>` is superfluous.

Reply:
Is it an option to move the instance definitions into `LogicTasks.Debug`? They will then only be loaded when `LogicTasks.Debug` is imported (which should not happen when using it as a library). Only thing I would have to change is to also expose the `Delayed` constructor in `Formula.Parsing.Delayed`.

Reply:
Yes, moving the (simplified, without `Show a` and `Pretty a` contexts) instances to the debug module sounds reasonable.

About the need to expose the `Delayed` constructor: In order to not have this exposure impact the external API, the following approach should be applicable.
- move `Delayed` from `Formula.Parsing.Delayed` to a new module `Formula.Parsing.Delayed.Internal`
- in `Formula.Parsing.Delayed`, import `Formula.Parsing.Delayed.Internal`, but export the type without its constructor
- in `Debug`, import `Formula.Parsing.Delayed.Internal`
- do not expose `Formula.Parsing.Delayed.Internal` to outside this library

Reply:
The instance for `Pretty` should be omitted completely. Since it is anyway only going to use the `Text.PrettyPrint.Leijen.Text.Pretty` instance for `String`, the user might as well use `Manual show` at that point.